.circleci/config.yml

```diff
@@ -36,12 +36,10 @@ aliases:
         # Install moreutils so we can use `ts` and `mispipe` in the following.
         sudo apt-get install -y moreutils
 
-        # CircleCI sets the following in Git config at clone time:
-        #   url.ssh://git@github.com.insteadOf https://github.com
+        # We use CircleCI's special step, checkout, to clone source
+        # code. It configures Git to use SSH.
         # This breaks the Git clones in the NVM `install.sh` we run
         # in `install-node`.
-        # TODO: figure out why that breaks, and whether we want it.
```

Comment:
The question still remains unsolved: why does CircleCI forcing ssh breaks NVM's `install.sh`?

Reply:
I think the reason is because the CircleCI's instance doesn't have an ssh public key. This hypothesis can be tested by adding `ssh-keygen -t ed25519 -f ~/.ssh/id_ed25519 -q -N ""` into config.yml.

Reply:
Surprisingly, node installs without any errors on my setup, without having to remove the git config.

![node](https://user-images.githubusercontent.com/58626718/105135325-d9b6aa80-5b15-11eb-84f1-c1e57f07e30f.png)

https://github.com/ganpa3/zulip/commit/5f5cae686e1c56c3dca0ab2577e9aef6e46759b0

Reply:
I tried forcing ssh for git on a 16.04 box, got error when nvm tried to `git clone`. @ganpa3 can you post the CircleCI test link?

Reply:
https://app.circleci.com/pipelines/github/ganpa3/zulip/21/workflows/c9759cce-630f-4cfa-9ce6-e610bba9fada/jobs/123

Reply:
puppeter tests are failing, though they are unrelated. They are failing on zulip/zulip too (https://github.com/zulip/zulip/commit/84a7e9f8377a9c0da26807995a5fc61eb3e91482)

Reply:
This is the commit that removed the .gitconfig: https://github.com/zulip/zulip/commit/9ce080ebb17f21fb67262644e79a80010ee44134 (2017). Maybe the error is no longer relevant for Ubuntu 18.04 and 20.04.